### PR TITLE
[wip] Phantom tests

### PR DESCRIPTION
**Pull request by bollwyvl**
This is for #28.

Still encountering some issues with anaconda-build environment.

### CODE REVIEW

**Comment by bollwyvl** [comment]
I am perilously close to getting the casper tests to run on anaconda cloud.

http://anaconda.org/nbcio/nbpresent/builds/85/1#L2457

it gets all the way to actually running the casper command and:
```
/opt/miniconda/conda-bld/test-tmp_dir/node_modules/phantomjs/lib/phantom/bin/phantomjs: error while loading shared libraries: libfreetype.so.6: cannot open shared object file: No such file or directory
```
If anyone (cc @mmarchetti @malev @damianavila @ijstokes) has any insight as to how it is best to ensure such libraries are available in the test environment, that would be extremely helpful. yum?

haven't tried the docker thing yet to see if that gives me more insight. it took me a while to realize how "clean" the environment is in which the tests are run, which made even starting stuff complex. adding a second package.json just for testing in conda.recipe seems to be working pretty well... and is good from a reuse/template point of view... i confess magic file names, nearly-duplicate keys in meta.yml, etc. are a little perplexing at times.

**Comment by damianavila** [comment]
>If anyone (cc @mmarchetti @malev @damianavila @ijstokes) has any insight as to how it is best to ensure such libraries are available in the test environment, that would be extremely helpful. yum?

Did you try adding phantomjs into the `test: requires:` field in the conda recipe?

**Comment by bollwyvl** [comment]
Heh, not looking so good:
http://anaconda.org/search?q=phantomjs

Nobody has a solid offering across all the platforms, and nobody has osx.

The (probably) last casperjs wants 1.9.8 i think: Nightmarejs has been
recommended by the maintainer:
http://www.nightmarejs.org/

(it does look very nice... es6 generators and all that)

But I *REALLY* want to be able to reuse the upstream tooling:
https://github.com/jupyter/notebook/blob/master/notebook/tests/util.js#L640

Anyhow... I'll try with 1.9.7 (tonight) and see how that goes!

**Comment by bollwyvl** [comment]
Hooray, working build!

Still a lot more to test, but will continue to make progress!

**Comment by damianavila** [comment]
>Hooray, working build!

Great!

File: nbpresent/install.py [hunk 1]
```diff
@@ -62,13 +63,13 @@ def install(enable=False, **kwargs):
...
         pprint(cm.get("jupyter_notebook_config"))
 
-        try:
-            subprocess.call(["conda", "info", "--root"])
-            print("conda detected")
-            _jupyter_config_dir = ENV_CONFIG_PATH[0]
-        except OSError:
-            print("conda not detected")
-            _jupyter_config_dir = jupyter_config_dir()
+        _jupyter_config_dir = jupyter_config_dir()
```
Comment (bollwyvl) [inline nbpresent/install.py]: TODO: fix this with nbsetuptools

**Comment by bollwyvl** [comment]
if the :green_heart: stays on, this is ready to go... i'm going to jump back on binder, and start getting that ready to go.

i did have to revert some of the setup stuff @damianavila, but when we get it working with nbsetuptools, we can address it.

**Comment by damianavila** [comment]
>i did have to revert some of the setup stuff @damianavila, but when we get it working with nbsetuptools, we can address it.

**Pull request merged**